opusfilter/opusfilter.py

```diff
@@ -71,6 +74,108 @@ def dict_set(key, value, dictionary):
+
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            return self.parallelize(*args, **kwargs)
+        return wrapper
+
+    @staticmethod
+    def _split(infiles, outfiles, n_jobs):
+        """split files into parts, and write them to temporary files"""
+        chunk_size = int(math.ceil(count_lines(infiles[0]) / n_jobs))
+
+        in_chunked_files = []
+        out_chunked_file = []
+        infileobjs = [file_open(infile) for infile in infiles]
+        for i, lines in enumerate(zip(*infileobjs)):
+            if i % chunk_size == 0:
+                intmpfiles = [f"{fname}.{str(i // chunk_size)}" for fname in infiles]
+                in_chunked_files.append(intmpfiles)
+                outtmpfiles = [f"{fname}.{str(i // chunk_size)}" for fname in outfiles]
+                out_chunked_file.append(outtmpfiles)
+                intmpfiles_objs = [file_open(fname, mode="w") for fname in intmpfiles]
+
+            for line, tmpfile in zip(lines, intmpfiles_objs):
+                tmpfile.write(line)
+        for tmpfile in intmpfiles_objs:
+            tmpfile.close()
+        return in_chunked_files, out_chunked_file
+
+    @staticmethod
+    def _merge(in_chunked_files, outfiles, out_chunked_files, limit):
+        """merge temporary files into final files and delete temporary files"""
+        for outfile, parts in zip(outfiles, zip(*out_chunked_files)):
+            with file_open(outfile, 'w') as out:
+                for i, line in enumerate(fileinput.input(parts)):
+                    out.write(line)
+                    if limit and i >= limit - 1:
+                        break
+            for part in parts:
+                os.remove(part)
+        for parts in in_chunked_files:
+            for part in parts:
+                os.remove(part)
+
+    def parallelize(self, obj, parameters, overwrite=False):
+        """Wrapper for parallelizing a function"""
+        # check if parameters are valid
+        if not isinstance(obj, OpusFilter):
+            raise TypeError("Decorator parallel can only be used with functions in OpusFilter.")
+        if "inputs" not in parameters:
+            obj.extra_parameters.update({"inputs"})
+        if "outputs" not in parameters and "output" not in parameters:
+            obj.extra_parameters.update({"outputs"})
+        n_jobs = parameters.pop('n_jobs', 1)
+        if n_jobs <= 1:
+            self.func(obj, parameters, overwrite)
+            return
+        # pylint: disable=W0212
+        obj._check_extra_parameters(self.extra_parameters, parameters)
+        infiles = [os.path.join(obj.output_dir, fname) for fname in parameters['inputs']]
+        if "outputs" in parameters:
+            outfiles = [os.path.join(obj.output_dir, fname) for fname in parameters['outputs']]
+        elif "output" in parameters:  # function `score` use `output` instead of `outputs`
+            outfiles = [os.path.join(obj.output_dir, parameters['output'])]
```

Comment:
Adding the `output_dir` to the filenames here seems to cause problems: I get the splitted files under `output_dir/output_dir`, where they can't be found by `_merge`. Note that `filter_data`, `score_data`, and `preprocess` are already adding the directory.